### PR TITLE
NPSP Potential Duplicates

### PR DESCRIPTION
**Pull request by npsp-reedestockton**
# Critical Changes

# Changes
Created NPSP Potential Duplicates component as a replacement for the standard Potential Duplicates component on the Contact page. Directs user to the NPSP Contact Merge page to review, select, and merge potential duplicate contacts.

# Issues Closed

# Community Ideas Delivered
[Potential Duplicates View Link doesn't point to the NPSP Merge Tool](https://ideas.salesforce.com/s/idea/a0B8W00000GdkqjUAB/potential-duplicates-view-link-doesnt-point-to-the-npsp-merge-tool)

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata
- Apex Classes
  - PotentialDuplicates
  - PotentialDuplicates_TEST
- LWC
  - potentialDuplicates.css
  - potentialDuplicates.html
  - potentialDuplicates.js
- Labels
  - potentialDuplicatesFoundNone
  - potentialDuplicatesFoundOne
  - potentialDuplicatesFoundMultiple

# Deleted Metadata

### CODE REVIEW

**Comment by npsp-reedestockton** [comment]
@lparrott This is a POC only. There is obvious work to be done with a label or two, error handling, cosmetics, and I'm sure other stuff as well. It won't be helpful to point out the small stuff, but significant concerns are a different story. Please call those out. For now, the link just redirects to the Contact Merge page with a parameter. Nothing has been done with the Contact Merge page itself to understand what it should do with the query parameter.

File: force-app/main/default/classes/PotentialDuplicates.cls [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        return duplicateCount;
+    }
+
+    private static Integer getDuplicateCount(Id recordId) {
+        Integer duplicateCount = 0;
+
+        List<Datacloud.FindDuplicatesResult> duplicateResults =
+                Datacloud.FindDuplicatesByIds.findDuplicatesByIds(new List<Id>{recordId});
+        if (duplicateResults?.size() > 0) {
+            duplicateCount = duplicateResults[0].getDuplicateResults()[0].getMatchResults()[0].size;
```
Comment (lparrott) [inline force-app/main/default/classes/PotentialDuplicates.cls:24]: If there are multiple Duplicate Rules setup for this object, only the first will be checked.

File: force-app/main/default/lwc/potentialDuplicates/potentialDuplicates.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<template>
+    <template lwc:if={isEnabled}>
```
Comment (lparrott) [inline force-app/main/default/lwc/potentialDuplicates/potentialDuplicates.html:2]: Replacing `isEnabled` with `displayCard` (or setting it to the same value) should hide the component and allow the Toast message to still display.

File: force-app/main/default/lwc/potentialDuplicates/potentialDuplicates.js-meta.xml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <description>Warns of potential duplicates as a result of Contact creation or edit.</description>
+    <isExposed>true</isExposed>
+    <masterLabel>NPSP Potential Duplicates</masterLabel>
+    <targets>
+        <target>lightning__RecordPage</target>
+    </targets>
+    <targetConfigs>
+        <targetConfig targets="lightning__RecordPage">
+            <objects>
+                <object>Contact</object>
```
Comment (lparrott) [inline force-app/main/default/lwc/potentialDuplicates/potentialDuplicates.js-meta.xml:13]: Good catch here!

File: force-app/main/default/classes/PotentialDuplicates.cls-meta.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ApexClass xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>53.0</apiVersion>
```
Comment (lparrott) [inline force-app/main/default/classes/PotentialDuplicates.cls-meta.xml:3]: We might as well raise these to the latest released version.

**Review by lparrott — COMMENTED** [review]
This is a great start! Looking forward to seeing this component get fleshed out.

**Comment by npsp-reedestockton** [comment]
This work is no longer a POC. It is fully functional, so converting this PR from Draft to Ready for Review.

**Comment by npsp-reedestockton** [comment]
[W-14252780](https://gus.lightning.force.com/lightning/r/a07EE00001bzUTPYA2/view)
[W-14252791](https://gus.lightning.force.com/lightning/r/a07EE00001byzxEYAQ/view)

**Review by m-kinkade — APPROVED** [review]
Looks great!

**Pull request merged**